users/models.py

```diff
@@ -0,0 +1,69 @@
+from django.db import models
+from django.utils import timezone
+from django.contrib.auth.models import (AbstractBaseUser,
+                                        PermissionsMixin,
+                                        BaseUserManager)
+
+
+class UserManager(BaseUserManager):
+
+    def create_user(self, email, first_name,
+                    last_name, birth_date, password,
+                    **other_fields):
+
+        if not email:
```

Comment:
From what I read about REQUIRED_FIELDS, it's only used to set which fields the "create_superuser" function will ask for,
So you still have to check the fields in the "create_user" function.

Some references:
https://github.com/django/django/blob/main/docs/topics/auth/customizing.txt - lines 569 to 594
https://docs.djangoproject.com/en/3.1/topics/auth/customizing/

Maybe I'm getting it wrong, ill wait for your reply before changing those lines.

Reply:
Or do you mean because blank=False is the default model option, there's no need to check them in "create_user"?

Reply:
You are correct about the REQUIRED_FIELDS.
What you can do is add null=true and blank=true to any required fields in the model, which will then prevent you from adding them in. This is preferred so that you don't have to check if the required fields are populated each time you change the model.
https://docs.djangoproject.com/en/3.1/ref/models/fields/

Reply:
If I'll set the fields to blank=true, won't it make it so they won't be required when creating a user? I read in the link you sent that `blank=true` will make it so the field won't be required in a from, do you want the fields to be non-mandatory?

Reply:
My mistake.  I meant null=False, blank=False.

Reply:
Hey @simzacks, should I get rid of the if statements in the create_user and create_superuser altogether? the reason I added them in the first place is because I was following a tutorial video on how to set up a custom user and in the video the commentator recommended adding if statements to later have easier time writing tests for the user. If it's better to have no if statements, I'll remove them. but then what exception should I expect in the tests?